python/lsst/ip/isr/isrTask.py

```diff
@@ -821,12 +833,25 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
 
         # Broken: DM-17152
         # Fringes are not tested to be handled correctly by Gen3 butler.
-        # inputs['fringes'] = (self.fringe.readFringes(inputsIds['ccdExposure'],
-        #                                                 assembler=self.assembleCcd
-        #                                                 if self.config.doAssembleIsrExposures else None)
-        #                         if self.config.doFringe and
-        #                            self.fringe.checkFilter(inputs['ccdExposure'])
-        #                         else pipeBase.Struct(fringes=None))
+        if self.config.doFringe is True and self.fringe.checkFilter(inputs['ccdExposure']):
+            inputs['fringes'] = self.fringe.loadFringes(inputs['fringes'],
+                                                        assembler=self.assembleCcd
+                                                        if self.config.doAssembleIsrExposures else None)
+        else:
+            inputs['fringes'] = pipeBase.Struct(fringes=None)
+
+        if self.config.doStrayLight is True and self.strayLight.checkFilter(inputs['ccdExposure']):
+            # This holds a deferred reference to the straylight file.
+            # However, these are packed oddly, so replace the data
+            # reference with a URI that can be handled by the
+            # straylight correction.  This breaks IO purity.
+            # straylightRef = inputs['strayLightData']
+            # diskUri = straylightRef.getUri()
+            # inputs['strayLightData'] = diskUri
+            print("Recieved straylightData: ", inputs['strayLightData'])
```

Comment:
I think you dont want to print here, but add it to inputs?

Reply:
As above, rewritten.  inputs['strayLightData'] should get what it says from the butler, and if not, rely on the straylight task to raise if it can't handle things.